node_modules/col-whiteboards/tests/test-whiteboards.js

```diff
@@ -450,6 +450,111 @@ describe('Whiteboards', function() {
+      });
+      return callback();
+    });
+  };
+
+  /**
+   * Test that verifies that whiteboards can be searched through
+   */
+  it('can be searched through', function(callback) {
+    setUpUsers(function(users, course, otherCourse) {
+
+      // Two users generate many inital whiteboards to test paging limit
+      TestsUtil.generateTestWhiteboards(users.regularA.client, course, 12, function(whiteboardsA) {
+        TestsUtil.generateTestWhiteboards(users.regularB.client, course, 12, function(whiteboardsB) {
+
+          // First user creates a private whiteboard
+          WhiteboardsTestUtil.assertCreateWhiteboard(users.regularA.client, course, 'Private Berkeley board', [users.regularA.me.id], function(privateBerkeleyBoard) {
+            whiteboardsA.push(privateBerkeleyBoard);
+
+            // Second user creates a private whiteboard
+            WhiteboardsTestUtil.assertCreateWhiteboard(users.regularB.client, course, 'Private Davis board', [users.regularB.me.id], function(privateDavisBoard) {
+              whiteboardsB.push(privateDavisBoard);
+            
+              // Each user creates one whiteboard shared with the other
+              WhiteboardsTestUtil.assertCreateWhiteboard(users.regularA.client, course, 'Shared Berkeley board', [users.regularA.me.id, users.regularB.me.id], function(sharedBerkeleyBoard) {
+                WhiteboardsTestUtil.assertCreateWhiteboard(users.regularB.client, course, 'Shared Davis board', [users.regularA.me.id, users.regularB.me.id], function(sharedDavisBoard) {
+                  var whiteboardsShared = [sharedBerkeleyBoard, sharedDavisBoard];
+
+                  // When no search filters are specified, a regular user should see all their whiteboards
+                  verifySearch(users.regularA.client, course, null, whiteboardsA.concat(whiteboardsShared), function() {
+                    verifySearch(users.regularB.client, course, null, whiteboardsB.concat(whiteboardsShared), function() {
+
+                      // When keyword filters are applied, a regular user should see only their whiteboards with matching titles
+                      verifySearch(users.regularA.client, course, {'keywords': 'Berkeley'}, [privateBerkeleyBoard, sharedBerkeleyBoard], function() {
+                        verifySearch(users.regularB.client, course, {'keywords': 'Davis'}, [privateDavisBoard, sharedDavisBoard], function() {
+                          verifySearch(users.regularA.client, course, {'keywords': 'Davis'}, [sharedDavisBoard], function() {
+                            verifySearch(users.regularB.client, course, {'keywords': 'Berkeley'}, [sharedBerkeleyBoard], function() {
+
+                              // An administrator should be able to see all whiteboards associated with a user
+                              verifySearch(users.instructor.client, course, {'user': users.regularA.me.id}, whiteboardsA.concat(whiteboardsShared), function() {
```

Comment:
This implies that the search would return the 12 whiteboards that were created first and the new whiteboard, which means there would be 13 results. However, as we're not specifying a total number of results to return, we only expect 10.

This should be updated to verify that the search is returning the correct number of results as well.